### ISSUE TITLE
Cannot upgrade to Magento 2.0.4

### ISSUE THREAD

**Issue by jaimestuardo** [magento/magento2#4013]
When Trying to upgrade from 2.0.2 to 2.04 using system upgrade, I get the following errors in Readiness Check:

Check Cron Scripts

Cron script readiness check failed. Hide detail

Error from Updater Application Cron Script:
Found non-writable path(s):
/home/depasa/public_html/app/design/frontend/Desytec/depasa_v1/Magento_Checkout/web/template/cart/totals
/home/depasa/public_html/webpay/cgi-bin/maestros/privada.pem
/home/depasa/public_html/webpay/cgi-bin/maestros/tbk_public_key.pem
/home/depasa/public_html/webpay/cgi-bin/maestros/backup
/home/depasa/public_html/webpay/cgi-bin/log/backup
/home/depasa/public_html/webpay/cgi-bin/datos/tbk_param_20160324.txt
/home/depasa/public_html/pub/static/frontend/Desytec/depasa_v1/es_ES/Magento_Checkout/js/form/element/region.js

For additional assistance, see cron scripts help.
Check Component Dependency

We found conflicting component dependencies. Hide detail

For additional assistance, see component dependency help .

All of them are nonsenses and help does not really help.

Regards
Jaime

**Comment by mazhalai** [comment]
@jaimestuardo so as suggested, did you check the file permissions? Looks like `pub/static` needs to be writable.  Not sure about the other files.

**Comment by jaimestuardo** [comment]
Of course. all permissions are correct, unless I have to set 777 permissions for all files and folders. Furthermore, what about app/design/frontend/Desytec/depasa_v1/Magento_Checkout/web/template/cart/totals? why that folder needs to be writable? it is my custom template.

And /webpay folder is a custom folder I am using with my own payment method. It contains CGI files that of course should not be writable. Is this a bug in cron validation which is validating something that it should not validate?

**Comment by mazhalai** [comment]
Yes I agree about the unnecessary files. Created MAGETWO-51544 to investigate and fix.

**Comment by jaimestuardo** [comment]
Meanwhile, is there a way to upgrade by bypassing that verification?

**Comment by mazhalai** [comment]
@jamescowie I would suggest to either move the files out of magento root during upgrade, and then copy them back or give full write permissions(not recommended) but as long as you change it back as soon as upgrade is done.

**Comment by andparsons** [comment]
I'm getting the same error while trying to upgrade via the web interface.
I've moved the custom theme outside the root folder, but it seems to have cached the error as it still shows the same paths are an issue.
I've tried clearing the cache from the flush cache page, but to no avail.
Can you help me find where this would be being cached?

**Comment by mazhalai** [comment]
@andparsons there are a number of files created in the var directory that begin with `.` check `.update_cronjob_status` and remove it. Wait for cron to run again, or run the commands manually, and you should be able to continue.

**Comment by jaimestuardo** [comment]
This is very annoying. I have removed the cache after I move the folders out of Magento root, and I got more errors now: PHP Version Check, PHP Settings Check, etc and etc. All those checks were correct before, and in fact, all of them are correct. I have removed .setup_cronjob_status and .update_cronjob_status according to this page http://devdocs.magento.com/guides/v2.0/comp-mgr/trouble/cman/component-depend.html but nothing works now.

**Comment by mazhalai** [comment]
@jamescowie can you please very that are your cronjobs running as expected?  It should create new `.setup_cronjob_status` and `.update_cronjob_status` files.

**Comment by jaimestuardo** [comment]
Only .update_cronjob_status:

-rw-r--r-- 1 depasa depasa    148 abr 11 19:09 .update_cronjob_status

**Comment by mazhalai** [comment]
@jaimestuardo you need a total of [3 cronjobs](http://devdocs.magento.com/guides/v2.0/config-guide/cli/config-cli-subcommands-cron.html#config-cli-cron-bkg) `/bin/magento setup:cron:run` creates `. setup_cronjob_status` Can you verify it is running? You can try running the command manually too.

**Comment by jaimestuardo** [comment]
Now, this is the error I am getting:

PHP Settings Check
Need Help?
PHP Documentation
Your PHP Version is 5.6.20, but always_populate_raw_post_data = 0. $HTTP_RAW_POST_DATA is deprecated from PHP 5.6 onwards and will be removed in PHP 7.0. This will stop the installer from running. Please open your php.ini file and set always_populate_raw_post_data to -1. If you need more help please call your hosting provider.

This is nonsense,since my php.ini has already that option with -1.

**Comment by piotrekkaminski** [comment]
This should be fixed in the upcoming 2.1 version. please try on the develop branch.

**Comment by rafaelstz** [comment]
Did not be fixed in 2.1 CE
```

Error from Updater Application Cron Script:

Found non-writable path(s):
/www/www.mystore.com/html/app/design/adminhtml/Magento
/www/www.mystore.com/html/app/design/frontend/Magento
```

Is necessary to create these empty folders to continue.
